### PR TITLE
Validate stride/padding/kernel_size length in slow_conv3d

### LINKED ISSUE #121095: Segmentation Fault Due to Empty Stride Argument in `torch._C._nn.slow_conv3d`
### 🐛 Describe the bug

The `torch._C._nn.slow_conv3d` function results in a segmentation fault when called with empty `stride`.

This is a fuzzed test input.

```bash
>>> import torch;torch._C._nn.slow_conv3d(torch.rand([9]), bias=torch.rand([1,1]), kernel_size=[1], padding =[1], str
ide=[], weight=torch.rand([4]))
Segmentation fault (core dumped)
```

Minified repro
```python
import torch
torch._C._nn.slow_conv3d(torch.rand([9]), bias=torch.rand([1,1]), kernel_size=[1], padding =[1], str
ide=[], weight=torch.rand([4]))
```

### Versions

PyTorch version: 2.3.0.dev20240220+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.3 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.8.17 (default, Jul  5 2023, 21:04:15)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-6.5.0-15-generic-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      48 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             256
On-line CPU(s) list:                0-255
Vendor ID:                          AuthenticAMD
Model name:                         AMD EPYC 7763 64-Core Processor
CPU family:                         25
Model:                              1
Thread(s) per core:                 2
Core(s) per socket:                 64
Socket(s):                          2
Stepping:                           1
Frequency boost:                    enabled
CPU max MHz:                        3529.0520
CPU min MHz:                        1500.0000
BogoMIPS:                           4890.76
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin brs arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
Virtualization:                     AMD-V
L1d cache:                          4 MiB (128 instances)
L1i cache:                          4 MiB (128 instances)
L2 cache:                           64 MiB (128 instances)
L3 cache:                           512 MiB (16 instances)
NUMA node(s):                       2
NUMA node0 CPU(s):                  0-63,128-191
NUMA node1 CPU(s):                  64-127,192-255
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Mitigation; safe RET
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] pytorch-triton==3.0.0+901819d2b6
[pip3] torch==2.3.0.dev20240220+cu118
[pip3] torchaudio==2.2.0.dev20240220+cu118
[pip3] torchvision==0.18.0.dev20240220+cu118
[conda] numpy                     1.23.5                   pypi_0    pypi
[conda] pytorch-triton            3.0.0+901819d2b6          pypi_0    pypi
[conda] torch                     2.3.0.dev20240220+cu118          pypi_0    pypi
[conda] torchaudio                2.2.0.dev20240220+cu118          pypi_0    pypi
[conda] torchvision               0.18.0.dev20240220+cu118          pypi_0    pypi

### PR DESCRIPTION
**Pull request by qflen**
Fixes #121095.

`torch._C._nn.slow_conv3d` SIGSEGVs when `kernel_size`, `padding`, or `stride` is shorter than 3. `slow_conv3d_forward_out_cpu` indexes each at `[0..2]` before any validation, so a short `IntArrayRef` reads past its storage. `slow_conv3d_shape_check` then runs on the extracted garbage ints, so in non-crashing runs the bug surfaces instead as a misleading `"stride should be greater than zero"` error.

Add `TORCH_CHECK(... .size() == 3, ...)` for each at the entry of `slow_conv3d_forward_out_cpu` and `slow_conv3d_backward_out_cpu`, matching `NaiveConvolutionTranspose3d.cpp`.

Test in `TestNNDeviceType.test_slow_conv3d_empty_stride` covers the short-length case for each argument.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thx
